presto-main/src/main/java/com/facebook/presto/operator/scalar/DateTimeFunctions.java

```diff
@@ -953,6 +953,24 @@ public static long timeZoneHourFromTimestampWithTimeZone(@SqlType(StandardTypes.
         return extractZoneOffsetMinutes(timestampWithTimeZone) / 60;
     }
 
+    @Description("last day of the month that contains timestamp")
+    @ScalarFunction("last_day")
+    @SqlType(StandardTypes.DATE)
+    public static long lastDayOfMonthFromTimestampWithTimeZone(ConnectorSession session, @SqlType(StandardTypes.TIMESTAMP_WITH_TIME_ZONE) long timestampWithTimeZone)
+    {
+        DateTime dateTime = new DateTime(unpackChronology(timestampWithTimeZone).millis().getMillis(unpackMillisUtc(timestampWithTimeZone))).dayOfMonth().withMaximumValue();
+        return MILLISECONDS.toDays(dateTime.dayOfMonth().roundCeilingCopy().getMillis());
```

Comment:
This implementation is also wrong because the `DateTime` object that is created has incorrect zone because it's using the system default instead of the zone in the provided argument.

In addition, this code is suspicious in that `unpackChronology(timestampWithTimeZone).millis().getMillis(tsMillis)` is a no-op.